### ISSUE TITLE
Unnecessary warning when assigning to special variables (eg PS1)

### ISSUE THREAD

**Issue by andreaja** [koalaman/shellcheck#185]
Given the following source:

``` bash
__prompt_command() {
    __pwd="$PWD"
}

PROMPT_COMMAND=__prompt_command

PS1="\$__pwd\$ "
```

Shellcheck will incorrectly generate a warning:

``` bash
$ shellcheck example.sh

In example.sh line 2:
    __pwd="$PWD"
    ^-- SC2034: __pwd appears unused. Verify it or export it.
```
